tests/regression_tests/benchmark/test_model_performance.py

```diff
@@ -0,0 +1,111 @@
+import os
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+
+import pytest
+from dataclasses_json import dataclass_json
+
+from ludwig.benchmarking.benchmark import benchmark
+from ludwig.constants import MODEL_ECD, MODEL_GBM
+from ludwig.utils.data_utils import load_json, load_yaml
+
+BENCHMARKING_CONFIG = """
+experiment_name: {experiment_name}
+hyperopt: false
+export:
+  export_artifacts: true
+  export_base_path: {export_base_path}
+profiler:
+  enable: false
+  use_torch_profiler: false
+  logging_interval: 0.1
+experiments:
+  - dataset_name: {dataset}
+    config_path: {config_path}
+"""
+
+
+class MetricRegressionDirection(Enum):
+    """Which direction is considered a regression."""
+
+    LOWER = -1
+    HIGHER = 1
+
+
+@dataclass_json
+@dataclass
+class ExpectedMetric:
+    # Output feature name.
+    output_feature_name: str
+
+    # Metric name.
+    metric_name: str
+
+    # Expected metric value.
+    expected_value: Union[int, float]
+
+    # Which direction is considered a regression.
+    regression_direction: int
+
+    # The percentage change that would trigger a notification/failure.
+    percent_change_sensitivity: float
+
+    def __post_init__(self):
+        if self.regression_direction == "LOWER":
+            self.regression_direction = -1
+        elif self.regression_direction == "HIGHER":
+            self.regression_direction = 1
+        else:
+            raise ValueError(
+                "Regression direction in the expected performance YAML file should be one" "of 'LOWER', 'HIGHER'."
+            )
+
+
+@pytest.mark.benchmark
+@pytest.mark.parametrize("model_type", [MODEL_GBM, MODEL_ECD])
+@pytest.mark.parametrize("dataset", ["ames_housing", "mercedes_benz_greener", "adult_census_income", "sarcos"])
```

Comment:
If we can standardize on a config name like `<dataset>.<descriptors>.yaml`, then we can get the dataset name from the config with `path.split('.')[0]`.

Then, by parameterizing over listdir, users could add new benchmarking configs without touching this file whatsoever.

Here's one idea:

```python
def get_dataset_from_benchmark_config_path(config_path: str) -> str:
    """path/to/config/<dataset>.<descriptors>.yaml -> dataset"""
    return os.path.basename(config_path).split(".")[0]


def assert_no_regression(expected_metric: ExpectedMetric, test_statistic: Union[int, float], dataset: str) -> None:
    """Assert that the test statistic has not regressed from the expected metric."""
    tolerance = (
            expected_metric.percent_change_sensitivity
            * expected_metric.expected_value
        )
    if expected_metric.regression_direction == MetricRegressionDirection.LOWER.value:
        assert test_statistic >= expected_metric.expected_value - tolerance, f"The observed '{expected_metric.metric_name}': {test_statistic} for output feature '{expected_metric.output_feature_name}' was regressively lower than the expected value '{expected_metric.expected_value}' with tolerance '{expected_metric.percent_change_sensitivity}'."
    else:
        assert test_statistic <= expected_metric.expected_value + tolerance, f"The observed '{expected_metric.metric_name}': {test_statistic} for output feature '{expected_metric.output_feature_name}' was regressively higher than the expected value '{expected_metric.expected_value}' with tolerance '{expected_metric.percent_change_sensitivity}'."


@pytest.mark.parametrize("config_path", os.listdir("tests/regression_tests/benchmark/configs"))
def test_performance(config_path, tmpdir):
    if not config_path.endswith(".yaml"):
        # Skip non-YAML files.
        return True

    dataset = get_dataset_from_benchmark_config_path(config_path)
    benchmarking_config = {
  "hyperopt": False, 
  "experiment_name": "regression_test",
  "experiments": [
    {
      "dataset_name": dataset,
      "config_path": config_path,
    }
  ], 
  "export": {
    "export_base_path": tmpdir,
    "export_artifacts": True,
  }, 
  "profiler": {
    "enable": false, 
    "use_torch_profiler": false, 
    "logging_interval": 0.1,
  }
}

    benchmark.benchmark(benchmarking_config_dict=benchmarking_config)

    test_statistics = benchmark.get_test_statistics(benchmarking_config)

    expected_metrics_path = os.path.join("tests/regression_tests/benchmark/expected_metrics", os.path.basename(config_path))
    if not os.path.exists(expected_metrics_path):
        raise FileNotFoundError(f"No corresponding expected metrics found for benchmarking config '{config_path}'. Please add a new metrics yaml file '{expected_metrics_path}'. Suggested content:\n{ ...<would be helpful to print out a suggested expected metrics dict that can be easily copy-pasted into a new file>... }")
    
    expected_metrics: Dict = load_yaml(expected_metrics_path)
    expected_metrics: List[ExpectedMetric] = [
        ExpectedMetric.from_dict(expected_metric) for expected_metric in expected_metrics["metrics"]
    ]

    for expected_metric in expected_metrics:
        assert_no_regression(expected_metric, test_statistics[expected_metric.output_feature_name][expected_metric.metric_name])
```

In this block, I've also suggested more detailed logs when the file does not exist, as well as when we detect a regression, which seems like a nice convenience for users who want to add more configs, and since we're exporting benchmark results to a tmpdir (which is the right thing to do), and which will be hard to inspect after the test cleans up.

Reply:
Made suggested changes